packages/app/src/background/services/zkIdentity/index.ts

```diff
@@ -76,7 +78,17 @@ export default class ZkIdentityService implements IBackupable {
     return identity ? bigintToHex(identity.genIdentityCommitment()) : "";
   };
 
-  getConnectedIdentityData = async (): Promise<ConnectedIdentityMetadata | undefined> => {
+  getConnectedIdentityData = async (_: unknown, meta: IZkMetadata): Promise<ConnectedIdentityMetadata | undefined> => {
+    const identity = await this.getConnectedIdentity();
+
+    if (!identity || identity.metadata.host !== meta.urlOrigin) {
+      return undefined;
+    }
+
+    return this.getConnectedIdentityMetadata(identity.metadata);
+  };
```

Comment:
I think it is not a good design to have to so similar functions with different name synonyms `readConnectedIdentityData` and `getConnectedIdentityData`. 
I think we can distinguish the use of this function weather internally or externally though the provider by only one function. 

```suggestion
  getConnectedIdentityData = async (_: unknown, meta: IZkMetadata): Promise<ConnectedIdentityMetadata | undefined> => {
    const identity = await this.getConnectedIdentity();

    if (!identity) {
      return undefined;
    }

    // This case when called from the provider
    if (meta.urlOrigin) {
      if (identity.metadata.host !== meta.urlOrigin) {
        return undefined;
      }
    }

    return this.getConnectedIdentityMetadata(identity.metadata);
  };
```

Reply:
It's not possible to distinguish them using provider because user has access to this code. If it's in provider anyone can call this method with meta params and bypass this check.

Reply:
Actually, I can use one function because this one doesn't have identity commitment or other private data. In case if someone gets this data it would reveal anything.

Reply:
@0xmad Yup right. :ok_hand: